src/hooks/screen/useCreateChat.ts

```diff
@@ -44,33 +46,62 @@ const useCreateChat = () => {
       console.log(e, 'error on memberSchema');
     }
   };
-  const createChannelJson = (response, selectedUser) => {
+  const createChannelJson = (response, selectedUser, isAnonymous = false) => {
     const channelWithMember = {...response.channel, members: response.members};
+    let targetName =
+      selectedUser?.user?.username || selectedUser?.user?.name || selectedUser.username;
+    let targetImage =
+      selectedUser?.user?.profilePicture ||
+      selectedUser?.user?.image ||
+      selectedUser.profilePicture;
+
     const targetRawJson = {
       type: 'notification.message_new',
       cid: response?.channel?.id,
       channel_id: '',
       channel_type: 'messaging',
       channel: channelWithMember,
       created_at: response?.channel,
-      targetName: selectedUser?.user?.username || selectedUser?.user?.name || selectedUser.username,
-      targetImage:
-        selectedUser?.user?.profilePicture ||
-        selectedUser?.user?.image ||
-        selectedUser.profilePicture
+      targetName,
+      targetImage
+    };
+
+    let defaultAnonUserInfo: AnonUserInfoTs = {
+      anon_user_info_color_code: '',
+      anon_user_info_color_name: '',
+      anon_user_info_emoji_code: '',
+      anon_user_info_emoji_name: ''
     };
-    const chatData = {
+
+    if (isAnonymous) {
+      const {
+        anonUserInfoColorCode,
+        anonUserInfoColorName,
+        anonUserInfoEmojiCode,
+        anonUserInfoEmojiName,
+        channelImage,
+        channelName
+      } = getChannelListInfo(response?.channel, signedProfileId, anonProfileId);
+      defaultAnonUserInfo = {
+        anon_user_info_color_code: anonUserInfoColorCode,
+        anon_user_info_color_name: anonUserInfoColorName,
+        anon_user_info_emoji_code: anonUserInfoEmojiCode,
+        anon_user_info_emoji_name: anonUserInfoEmojiName
+      };
+
+      targetName = channelName;
+      targetImage = channelImage;
+    }
+
+    const chatData: MessageAnonymouslyData = {
       channel: response?.channel,
       members: response?.members,
+      ...defaultAnonUserInfo,
       appAdditionalData: {
         rawJson: targetRawJson,
         message: '',
-        targetName:
-          selectedUser?.user?.username || selectedUser?.user?.name || selectedUser.username,
-        targetImage:
-          selectedUser?.user?.profilePicture ||
-          selectedUser?.user?.image ||
-          selectedUser.profilePicture
+        targetName,
+        targetImage
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `createChannelJson` has been significantly modified to handle anonymous messages. The changes seem to be logically correct, but there are a few potential issues:

1. There is no null check for `response?.channel`. If `response?.channel` is null or undefined, it could lead to unexpected behavior or errors in the following lines where it's used.

2. The `created_at` field is being assigned `response?.channel`, which seems incorrect as per the context. It should probably be a timestamp or date value.

3. The `targetRawJson` object now includes `targetName` and `targetImage` fields without any values. These fields are later updated based on whether the message is anonymous or not. This could potentially lead to confusion and bugs in the future. It would be better to assign these fields when they are defined.

Here's how you can address these issues:

```diff
- const channelWithMember = {...response.channel, members: response.members};
+ if (!response?.channel) {
+   throw new Error('Response channel is missing');
+ }
+ const channelWithMember = {...response.channel, members: response.members};

- created_at: response?.channel,
+ created_at: new Date(), // or another appropriate timestamp

- const targetRawJson = {
-   type: 'notification.message_new',
-   cid: response?.channel?.id,
-   channel_id: '',
-   channel_type: 'messaging',
-   channel: channelWithMember,
-   created_at: response?.channel,
-   targetName,
-   targetImage
- };
...
- targetRawJson.targetName = targetName;
- targetRawJson.targetImage = targetImage;
+ const targetRawJson = {
+   type: 'notification.message_new',
+   cid: response?.channel?.id,
+   channel_id: '',
+   channel_type: 'messaging',
+   channel: channelWithMember,
+   created_at: new Date(), // or another appropriate timestamp
+   targetName: targetName,
+   targetImage: targetImage
+ };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->